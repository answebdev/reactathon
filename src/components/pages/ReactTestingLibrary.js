import React from 'react';
import { Helmet } from 'react-helmet';
import { Container } from 'react-bootstrap';

export default function ReactTestingLibrary({ text }) {
  const styles = {
    fontSize: '16px',
  };

  return (
    <div>
      <Helmet>
        <title>Reactathon | React Testing Library</title>
      </Helmet>
      <Container>
        <div style={{ marginBottom: '40px' }} className='App'>
          <h3 style={{ marginTop: '40px', marginBottom: '20px' }}>
            <strong>React Testing Library</strong>
          </h3>
          <div>
            <p style={styles}>
              View the video&nbsp;
              <a
                href='https://www.youtube.com/watch?v=YQLn7ycfzEo'
                rel='noopener noreferrer'
                target='_blank'
              >
                here
              </a>
              . See the article and code&nbsp;
              <a
                href='https://www.leighhalliday.com/introduction-react-testing-library'
                rel='noopener noreferrer'
                target='_blank'
              >
                here
              </a>
              .
            </p>
            <p style={styles}>
              Resources:
              <ul>
                <li>
                  Testing Library &nbsp;| &nbsp;
                  <a
                    href='https://testing-library.com/'
                    rel='noopener noreferrer'
                    target='_blank'
                  >
                    View
                  </a>
                </li>
                <li>
                  React Testing Library (NPM) &nbsp;| &nbsp;
                  <a
                    href='https://www.npmjs.com/package/@testing-library/react'
                    rel='noopener noreferrer'
                    target='_blank'
                  >
                    View
                  </a>
                </li>
                <li>
                  Jest-Dom (adds on different 'expects' we can use with Jest -
                  little helpers) &nbsp;| &nbsp;
                  <a
                    href='https://github.com/testing-library/jest-dom'
                    rel='noopener noreferrer'
                    target='_blank'
                  >
                    View
                  </a>
                </li>
              </ul>
            </p>
            <hr />
            <header>
              <h1 data-testid='h1tag' className='fancy-h1'>
                {text}
              </h1>
            </header>
          </div>
        </div>
      </Container>
    </div>
  );
}

// export default ReactTestingLibrary;
