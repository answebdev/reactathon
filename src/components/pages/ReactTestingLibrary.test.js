import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import ReactTestingLibrary from './ReactTestingLibrary';

// See also: https://kentcdodds.com/blog/common-mistakes-with-react-testing-library
// See also: https://www.robinwieruch.de/react-testing-library

// Call the 'cleanup' function after each test to clean things up.
afterEach(cleanup);

// Basic Render Test
describe('React Testing Library Page', () => {
  test('renders react testing library component', () => {
    render(<ReactTestingLibrary />);
    screen.debug();
  });
});

// Snapshot Test
it('renders', () => {
  const { asFragment } = render(<ReactTestingLibrary text='Hello!' />);
  expect(asFragment()).toMatchSnapshot();
});

it('inserts text in h1', () => {
  const { getByTestId, getByText } = render(
    <ReactTestingLibrary text='Hello!' />
  );

  expect(getByTestId('h1tag')).toHaveTextContent('Hello!');
  expect(getByText('Hello!')).toHaveClass('fancy-h1');
});
