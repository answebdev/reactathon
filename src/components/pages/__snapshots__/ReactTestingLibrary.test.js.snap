// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    <div
      class="container"
    >
      <div
        class="App"
        style="margin-bottom: 40px;"
      >
        <h3
          style="margin-top: 40px; margin-bottom: 20px;"
        >
          <strong>
            React Testing Library
          </strong>
        </h3>
        <div>
          <p
            style="font-size: 16px;"
          >
            View the video 
            <a
              href="https://www.youtube.com/watch?v=YQLn7ycfzEo"
              rel="noopener noreferrer"
              target="_blank"
            >
              here
            </a>
            . See the article and code 
            <a
              href="https://www.leighhalliday.com/introduction-react-testing-library"
              rel="noopener noreferrer"
              target="_blank"
            >
              here
            </a>
            .
          </p>
          <p
            style="font-size: 16px;"
          >
            Resources:
          </p>
          <ul>
            <li>
              Testing Library  |  
              <a
                href="https://testing-library.com/"
                rel="noopener noreferrer"
                target="_blank"
              >
                View
              </a>
            </li>
            <li>
              React Testing Library (NPM)  |  
              <a
                href="https://www.npmjs.com/package/@testing-library/react"
                rel="noopener noreferrer"
                target="_blank"
              >
                View
              </a>
            </li>
            <li>
              Jest-Dom (adds on different 'expects' we can use with Jest - little helpers)  |  
              <a
                href="https://github.com/testing-library/jest-dom"
                rel="noopener noreferrer"
                target="_blank"
              >
                View
              </a>
            </li>
          </ul>
          <p />
          <hr />
          <header>
            <h1
              class="fancy-h1"
              data-testid="h1tag"
            >
              Hello!
            </h1>
          </header>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
